closure/goog/array/array.js

```diff
@@ -346,20 +346,33 @@ goog.array.map = goog.NATIVE_ARRAY_PROTOTYPES &&
 goog.array.reduce = goog.NATIVE_ARRAY_PROTOTYPES &&
                     (goog.array.ASSUME_NATIVE_FUNCTIONS ||
                      goog.array.ARRAY_PROTOTYPE_.reduce) ?
-    function(arr, f, val, opt_obj) {
-      goog.asserts.assert(arr.length != null);
-      if (opt_obj) {
-        f = goog.bind(f, opt_obj);
+  function(arr, f, val, opt_obj) {
+    goog.asserts.assert(arr.length != null);
+    var params = [];
+    for(var i = 1, l = arguments.length; i < l; i++){
```

Comment:
Here's a couple pens that might help explain the issue more clearly:
http://codepen.io/moagrius/pen/RPqMVG?editors=001 (just inline comments since the lines here are getting crowded, no output)
http://codepen.io/moagrius/pen/jPQzrz?editors=001 (has a simple example showing the issue)